sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/LargeBlobTest.groovy

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob
+
+import com.azure.core.http.HttpPipelineCallContext
+import com.azure.core.http.HttpPipelineNextPolicy
+import com.azure.core.http.HttpResponse
+import com.azure.core.http.policy.HttpPipelinePolicy
+import com.azure.core.util.Context
+import com.azure.storage.blob.models.BlockListType
+import com.azure.storage.blob.models.ParallelTransferOptions
+import com.azure.storage.common.implementation.Constants
+import com.azure.storage.common.policy.StorageSharedKeyCredentialPolicy
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import spock.lang.Ignore
+import spock.lang.Requires
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicLong
+import java.util.function.BiFunction
+
+class LargeBlobTest extends APISpec {
+
+    long maxBlockSize =  4000L * Constants.MB
+    BlobClient blobClient
+    BlobAsyncClient blobAsyncClient
+    String blobName
+    List<Mono<Long>> putBlockPayloadSizes = Collections.synchronizedList(new ArrayList<>())
+    AtomicLong count = new AtomicLong()
+    boolean collectSize = true
+
+    def setup() {
+        blobName = generateBlobName()
+        def basic = cc.getBlobClient(blobName)
+        this.blobClient = getBlobClient(
+            primaryCredential,
+            basic.getBlobUrl(),
+            new PayloadDroppingPolicy(),
+            // Regenerate auth header after body got substituted
+            new StorageSharedKeyCredentialPolicy(primaryCredential)
+        )
+        this.blobAsyncClient = getBlobAsyncClient(
+            primaryCredential,
+            basic.getBlobUrl(),
+            new PayloadDroppingPolicy(),
+            // Regenerate auth header after body got substituted
+            new StorageSharedKeyCredentialPolicy(primaryCredential)
+        )
+    }
+
+    @Requires({ liveMode() })
+    @Ignore("Takes really long time")
+    def "Stage Real Large Blob"() {
+        given:
+        def stream = createLargeInputStream(maxBlockSize)
+        def blockId = Base64.getEncoder().encodeToString(UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8))
+        def client = cc.getBlobClient(blobName)
+
+        when:
+        client.getBlockBlobClient().stageBlock(blockId, stream, maxBlockSize)
+        client.getBlockBlobClient().commitBlockList([blockId])
+        def blockList = client.getBlockBlobClient().listBlocks(BlockListType.COMMITTED)
+
+        then:
+        blockList.committedBlocks.size() == 1
+        blockList.committedBlocks.get(0).getSizeLong() == maxBlockSize
+    }
+
+    @Requires({ liveMode() })
+    def "Stage Large Blob"() {
+        given:
```

Comment:
Is the only difference between this test and the last one that this one drops the payload and counts it in the pipeline? Can you leave a note about that?

Reply:
commented